core/engine/src/job.rs

```diff
@@ -133,6 +115,168 @@ impl NativeJob {
     }
 }
 
+/// The [`Future`] job returned by a [`NativeAsyncJob`] operation.
+pub type BoxedFuture<'a> = Pin<Box<dyn Future<Output = JsResult<JsValue>> + 'a>>;
+
+/// An ECMAScript [Job] that can be run asynchronously.
```

Comment:
Are there any scenarios where these are *not* ran asyncrhonously?

Reply:
Yeah, an implementor can decide to block on each future if they want to.